presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -151,6 +154,23 @@ public TableStatistics getTableStatistics(
         }
     }
 
+    private TableStatistics createZeroStatistics(Map<String, ColumnHandle> columns, Map<String, Type> columnTypes)
+    {
+        TableStatistics.Builder result = TableStatistics.builder();
+        result.setRowCount(Estimate.of(0));
+        columns.forEach((columnName, columnHandle) -> {
+            Type columnType = requireNonNull(columnTypes.get(columnName), "columnType is null");
+            ColumnStatistics.Builder columnStatistics = ColumnStatistics.builder();
+            columnStatistics.setNullsFraction(Estimate.of(0));
+            columnStatistics.setDistinctValuesCount(Estimate.of(0));
+            if (isDataSizeRequired(columnType)) {
```

Comment:
a bit misnomer, as nothing is _required_.
`!isFixedSize(type)`? `!variableSize(type)`?

Also, why have this `if` at all? What if the `columnStatistics.setDataSize(Estimate.of(0));` was unconditional?

Reply:
I agree. But if we no there is no data, we know more than usual. (I don't insist, just thinking loud)

Reply:
Maybe `hasDataSize()` would sound better?

Reply:
> I agree. But if we no there is no data, we know more than usual. (I don't insist, just thinking loud)

For all the fixed length types we can also set dataSize to `<single value size> * <number of values>`, but that's not what we do right now. If we decided to do this, we should change it in other places. I just want the statistics for zero partitions to be consistent with what we have for empty and non-empty partitions.